api/session.go

```diff
@@ -179,6 +180,8 @@ func (s *Session) RenewPeriodic(initialTTL string, id string, q *WriteOptions, d
 			// Attempt a session destroy
 			s.Destroy(id, q)
 			return nil
+		case <-ctx.Done():
```

Comment:
I generally treat a context cancellation as "stop doing anything", rather than "attempt to shut down cleanly", so this is intentional.